### ISSUE TITLE
Registration of filters on base classes no longer work

### ISSUE THREAD

**Issue by mciancia8** [autofac/Autofac.WebApi#45]
I have two filters set on two different base classes. I then have a controller (PersController) that derives from both of them.

```
containerBuilder.RegisterType<ExceptionFilter>()
	.As<IAutofacExceptionFilter>()
	.AsWebApiExceptionFilterFor<ApiController>()
	.InstancePerRequest();

containerBuilder.RegisterType<AuthenticationFilter>()
	.As<IAutofacAuthenticationFilter>()
	.AsWebApiAuthenticationFilterFor<AuthenticationControllerBase>()
	.InstancePerRequest();
	
public class PersController : AuthenticationControllerBase
{
}

public abstract class AuthenticationControllerBase : ApiController
{
}
```

With latest version (v4.3.0), if I call an API in PersController, I don't see either of these filters run. In version v4.2.1, this worked.

If I change the registration to the following, then it works:

```
containerBuilder.RegisterType<ExceptionFilter>()
	.As<IAutofacExceptionFilter>()
	.AsWebApiExceptionFilterFor<PersController>()
	.InstancePerRequest();

containerBuilder.RegisterType<AuthenticationFilter>()
	.As<IAutofacAuthenticationFilter>()
	.AsWebApiAuthenticationFilterFor<PersController>()
	.InstancePerRequest();
```

This is a huge change and I assume that this is a bug, not that this was an intended change.

**Comment by alistairjevans** [comment]
Yep, definitely a bug! Thanks for reporting this. I've got a PR raised to fix this and we'll get a 4.3.1 out soon. 

In the meantime, you can workaround this by using the new predicate-style mechanism, and checking yourself, like so:

```
containerBuilder.RegisterType<AuthenticationFilter>()
	.As<IAutofacAuthenticationFilter>()
	.AsWebApiAuthenticationFilterWhere(descriptor => typeof(AuthenticationControllerBase).IsAssignableFrom(descriptor.ControllerDescriptor.ControllerType))
	.InstancePerRequest();
```

**Comment by mciancia8** [comment]
thank you for the quick turnaround and for the workaround!

**Comment by tillig** [comment]
Resolved by #46.
